test/integration_test/common_test.go

```diff
@@ -440,16 +456,50 @@ func verifyScheduledNode(t *testing.T, appNode node.Node, volumes []string) {
 			}
 		}
 	}
+	return scores
+}
 
-	highScore := 0
-	for _, score := range scores {
-		if score > highScore {
-			highScore = score
-		}
+// Verify nodes are scheduled on appropriate nodes based on hyperconvergence scoring
+// This method assumes we have pod AntiAffinities set in the deployment to deploy each replica on a different node.
+func verifyScheduledNodesMultipleReplicas(t *testing.T, appNodes []node.Node, volumes []string) {
+	driverNodes, err := storkVolumeDriver.GetNodes()
+	require.NoError(t, err, "Error getting nodes from stork driver")
+
+	//The scores are stored in a map where the keys are node names, and the values are the corresponding scores.
+	scores := getScoringBasedOnHyperconvergence(t, driverNodes, volumes)
+	replicaCount := len(appNodes)
+
+	// The nodes are sorted based on their scores, from highest to lowest.
+	// From the nodeScores slice we want to select the top nodes with the highest scores.
+	// The actual nodes which the pods get scheduled on maybe different in case of multiple nodes having the same scores.
+	// However, the scheduled nodes should have the same scores as the top n highest scores where n is number of replicas.
+	nodeScores := descendingSortBasedOnValue(scores)
+	expectedNodeScores := nodeScores[0:replicaCount]
+	logrus.Infof("Expected nodescores are %v", expectedNodeScores)
+
+	//appNodeScores is a map of nodeName:score, in which we store the calculated scores of the set of scheduled nodes.
+	appNodeScores := make(map[string]int)
+	for _, value := range appNodes {
```

Comment:
nit.. Let's call it node or appNode instead of "value".

Reply:
will address in revision